packages/google_sign_in/google_sign_in_ios/ios/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -190,31 +199,38 @@ - (void)requestScopes:(nonnull NSArray<NSString *> *)scopes
   NSSet<NSString *> *requestedScopes = self.requestedScopes;
 
   @try {
-    [self.signIn addScopes:requestedScopes.allObjects
+    GIDGoogleUser *currentUser = self.signIn.currentUser;
+    if (currentUser == nil) {
+      completion(nil, [FlutterError errorWithCode:@"sign_in_required"
+                                          message:@"No account to grant scopes."
+                                          details:nil]);
+    }
+    [currentUser addScopes:requestedScopes.allObjects
         presentingViewController:[self topViewController]
-                        callback:^(GIDGoogleUser *addedScopeUser, NSError *addedScopeError) {
-                          BOOL granted = NO;
-                          FlutterError *error = nil;
-                          if ([addedScopeError.domain isEqualToString:kGIDSignInErrorDomain] &&
-                              addedScopeError.code == kGIDSignInErrorCodeNoCurrentUser) {
-                            error = [FlutterError errorWithCode:@"sign_in_required"
-                                                        message:@"No account to grant scopes."
-                                                        details:nil];
-                          } else if ([addedScopeError.domain
-                                         isEqualToString:kGIDSignInErrorDomain] &&
-                                     addedScopeError.code ==
-                                         kGIDSignInErrorCodeScopesAlreadyGranted) {
-                            // Scopes already granted, report success.
-                            granted = YES;
-                          } else if (addedScopeUser == nil) {
-                            granted = NO;
-                          } else {
-                            NSSet<NSString *> *grantedScopes =
-                                [NSSet setWithArray:addedScopeUser.grantedScopes];
-                            granted = [requestedScopes isSubsetOfSet:grantedScopes];
-                          }
-                          completion(error == nil ? @(granted) : nil, error);
-                        }];
+                      completion:^(GIDSignInResult *_Nullable signInResult,
+                                   NSError *_Nullable addedScopeError) {
+                        BOOL granted = NO;
+                        FlutterError *error = nil;
+
+                        if ([addedScopeError.domain isEqualToString:kGIDSignInErrorDomain] &&
+                            addedScopeError.code == kGIDSignInErrorCodeMismatchWithCurrentUser) {
+                          error =
+                              [FlutterError errorWithCode:@"request_scopes"
```

Comment:
We may want to give this a new code; currently string matching on codes is kind of the best option clients have for `PlatformException` handling (until we [fix it long term](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#platform-exception-handling)), so when we have a distinct case it's useful to give it its own code.